### PR TITLE
Fix analysis stop on a client exception

### PR DESCRIPTION
**Pull request by temanbrcom**
The client with an incorrect Zowe profile may produce an exception that may stop the analysis without any notifications. The exception should be caught explicitly.

Fix: #643

### CODE REVIEW

File: server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java
Comment (Nurkambay) [inline server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:151]: This method can be moved to `CopybookModel` class

Comment (grianbrcom) [reply server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:151]: Agree. Or we can have a constructor with only one parameter: copybookName.

Comment (temanbrcom) [reply server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:151]: That's true. But I think that due to `CopybookServiceImpl` is the only class that creates copybook models, an empty model is a part of its internal logic, not of `CopybookModel` itself.

**Comment by temanbrcom** [comment]
Waiting for Brian's confirmation to mark it "ready for review".

File: server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:123]: These multiple if can be merged like if, else if, else

File: server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:173]: Meaningful error message needed

File: server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java
Comment (asatklichov) [inline server/src/main/java/com/broadcom/lsp/cobol/service/CopybookServiceImpl.java:175]: No return in catch block, can  be taken out of catch

**Review by asatklichov — CHANGES_REQUESTED** [review]

**Comment by temanbrcom** [comment]
The new build produces shows this log (the part of the file is skipped):
```
06:49:44.570 [pool-3-thread-1] ERROR com.broadcom.lsp.cobol.service.CobolTextDocumentService - An exception thrown while applying analysis for file:///c%3A/Users/bj608554/Projects/COBOL/CAMRCOBR.CBL:
java.lang.InterruptedException: Parsing interrupted by user.
	at com.broadcom.lsp.cobol.core.annotation.HandleThreadInterruption.invoke(HandleThreadInterruption.java:44)
	at com.broadcom.lsp.cobol.core.preprocessor.delegates.GrammarPreprocessorListenerImpl.processCopybook(GrammarPreprocessorListenerImpl.java:323)
```
I think it is because of the `Thread.currentThread().interrupt()` in the catch block

**Comment by temanbrcom** [comment]
To be fixed by #675

**Pull request closed without merging**
